tiledb/sm/rest/rest_client.h

```diff
@@ -566,6 +572,10 @@ class RestClient {
    * @return Status
    */
   Status ensure_json_null_delimited_string(Buffer* buffer);
+
```

Comment:
My current working branch for `class RestClient` exposes the member variable `extra_headers_` with a constant accessor. This supplants the need for a friend declaration.

```suggestion
  /**
   * Constant accessor to `extra_headers_` member variable.
   */
  const std::unordered_map<std::string, std::string>& extra_headers() const {
    return extra_headers_;
  }
```

The reason for this is that the member function `StorageManager::tags` is removed (along with `set_tag`) in the branch. `extra_headers()` replaces it.